scripts/test_ansible_test_runner.py

```diff
@@ -0,0 +1,120 @@
+from unittest.mock import patch, mock_open
+import os
+
+import sys
+
+sys.path.insert(0, os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+from scripts.ansible_test_runner import find_pending_snapshot, export_component_variables, read_release_candidates, run_ansible_tests
+
+
+def test_find_pending_snapshot():
+    data = {
+        "snapshots": [
+            {"metadata": {"status": "successful", "generated_at": "2025-03-10T10:00:00.000000"}},
+            {"metadata": {"status": "pending", "generated_at": "2025-03-10T11:00:00.000000"}},
+            {"metadata": {"status": "failed", "generated_at": "2025-03-10T12:00:00.000000"}}
+        ]
+    }
+    
+    # shoudl find the second snapshot
+    timestamp, snapshot = find_pending_snapshot(data)
+    assert timestamp == "2025-03-10T11:00:00.000000"
+    assert snapshot["metadata"]["status"] == "pending"
+    
+    # no pending snapshots
+    data = {"snapshots": [{"metadata": {"status": "successful", "generated_at": "2025-03-10T10:00:00.000000"}}]}
+    timestamp, snapshot = find_pending_snapshot(data)
+    assert timestamp is None
+    assert snapshot is None
+
+def test_export_component_variables():
+    snapshot = {
+        "components": [
+            {
+                "repository": "kubernetes-sigs/cluster-api",
+                "ref": "v1.0.0",
+                "image_url": None
+            },
+            {
+                "repository": "openshift/assisted-service",
+                "ref": "abc123",
+                "image_url": "quay.io/edge-infrastructure/assisted-service:latest-abc123"
+            }
+        ]
+    }
+    
+    with patch.dict(os.environ, {}, clear=True):
+        export_component_variables(snapshot)
+        
+        assert os.environ["CAPI_VERSION"] == "v1.0.0"
+        assert os.environ["ASSISTED_SERVICE_IMAGE"] == "quay.io/edge-infrastructure/assisted-service:latest-abc123"
+
+def test_run_ansible_tests():
+    expected_cmd = [
+        "ansible-playbook",
+        "test/ansible/run_test.yaml",
+        "-i",
+        "test/ansible/inventory.yaml",
+    ]
+    
+    with patch('subprocess.run') as mock_run:
+        mock_run.return_value.returncode = 0
+        assert run_ansible_tests() is True
+        mock_run.assert_called_once_with(
+            expected_cmd, 
+            check=False, 
+            capture_output=False, 
+            text=True
+        )
+        
+    # failure case
+    with patch('subprocess.run') as mock_run:
+        mock_run.return_value.returncode = 1
+        assert run_ansible_tests() is False
+        mock_run.assert_called_once_with(
+            expected_cmd, 
+            check=False, 
+            capture_output=False, 
+            text=True
+        )
+
+def test_read_release_candidates_success():
+    yaml_content = """
+    snapshots:
+      - metadata:
+          generated_at: '2025-03-10T10:32:04.642635'
```

Comment:
we can leave generated_at, but I'd add a unique name/id (could be based on time, but since we have this field could be totally random now).